examples/livesync/components/Drawer/Comments.tsx

```diff
@@ -29,37 +30,42 @@ export const Comments = ({ issueId, user }: Props) => {
   } = useForm<FormData>();
 
   const onSubmit = async ({ content }: FormData) => {
-    if (issueId === null || !user) return;
+    if (issueId === null || !user || !model) return;
 
-    const newComment = await postComment({ userId: user.id, issueId, content });
-    setComments((prev) => [
-      {
-        ...newComment,
-        ...user,
-      },
-      ...(prev || []),
-    ]);
-    setValue('content', '');
-  };
+    const mutationID = uuidv4();
+    const data = {
+      userId: user.id,
+      issueId,
+      content,
+      mutationID,
+      first_name: user.first_name,
+      last_name: user.last_name,
+      color: user.color,
+      updated_at: new Date().toISOString(),
+    };
 
-  useEffect(() => {
-    setComments([]);
-    if (!issueId) return;
+    const [confirmation, cancel] = await model?.optimistic({
+      mutationID,
+      name: CommentsMergeEvent.POST_COMMENT,
+      data,
```

Comment:
I have created a ticket to support event type inference in the library: https://ably.atlassian.net/browse/COL-652